### DISCUSSION TITLE
Docu about Customize HTML output?

### DISCUSSION THREAD

**Discussion by wh81752** [Mogztter/asciidoctor-web-pdf#481] [Q&A] [answered]
@Mogztter - Is there some documentation around how to fully customize the generated HTML output?

**Comment by ggrossetie** [comment, marked as answer]
You have basically two levels of customization:

- Using a custom stylesheet with the `stylesheet` attribute
- Using a custom layout

When using a custom stylesheet, you will probably need to inspect the HTML produced by the built-in converter to understand how to customize it. I recommend to use the `--preview` option when working on a new design. See: https://github.com/Mogztter/asciidoctor-web-pdf/discussions/441

If you want to go deeper then you can provide a custom layout. You can read https://github.com/Mogztter/asciidoctor-web-pdf#custom-layout which briefly explains how to extend the built-in converter.
As mentioned in the README you will need JavaScript knowledge and you will also need to understand how Asciidoctor works (i.e., it creates an AST from an AsciiDoc document and then forwards it to a converter).

You can take a look at the built-in HTML 5 converter (written in Ruby): https://github.com/asciidoctor/asciidoctor/blob/master/lib/asciidoctor/converter/html5.rb

For each node type, Asciidoctor will call the corresponding method. For instance, to convert an Image to HTML, Asciidoctor will call `convert_image`, to convert a paragraph it will call `convert_paragraph` and so on...

**Reply by wh81752** [reply]
@Mogztter - thx a lot for this pointers, very welcomed.
